test/test_binary_ufuncs.py

```diff
@@ -268,6 +268,72 @@ def test_inplace_division(self, device):
         id_after = id(t)
         self.assertEqual(id_before, id_after)
 
+    @dtypes(*torch.testing.get_all_dtypes(include_bool=False, include_complex=False))
+    def test_div_rounding_modes(self, device, dtype):
+        if dtype.is_floating_point:
+            a = 10 * torch.randn(100, device=device, dtype=dtype)
+            # Avoid division by zero so we can test (a / b) * b == a
+            b = 0.1 + 2 * torch.rand(100, device=device, dtype=dtype)
+        else:
+            info = torch.iinfo(dtype)
+            low, high = info.min, info.max
+            a = torch.randint(low, high, (100,), device=device, dtype=dtype)
+            b = torch.randint(1, high, (100,), device=device, dtype=dtype)
+
+            # floor(a / b) * b can be < a, so fixup slightly to avoid underflow
+            a = torch.where(a < 0, a + b, a)
+
+        d_true = torch.divide(a, b, rounding_mode='true')
+        self.assertTrue(d_true.is_floating_point())
+        self.assertEqual(d_true * b, a.to(d_true.dtype))
+
+        d_floor = torch.divide(a, b, rounding_mode='floor')
+        if dtype != torch.bfloat16:
+            self.assertEqual(d_floor * b + torch.remainder(a, b), a)
+        else:
+            self.assertEqual(d_floor * b + torch.remainder(a.float(), b.float()), a,
+                             exact_dtype=False)
+
+        d_trunc = torch.divide(a, b, rounding_mode='trunc')
+        rounding_unsupported = (
+            dtype == torch.half and device != 'cuda' or
+            dtype == torch.bfloat16 and device != 'cpu')
+        d_ref = d_true.float() if rounding_unsupported else d_true.float()
+        self.assertEqual(d_trunc, d_ref.trunc().to(dtype))
+
+        # Compare against NumPy
+        if dtype.is_floating_point:
```

Comment:
For the NumPy comparison start a new test and set the default scalar type to double in a try/finally block and test uint8, int8, long, float32, and complex64 dtype division. Enumerate the modes on large random tensors (with zero divisors removed for the integer types), and then every other element of those tensors (similar to what you're doing below). 

This should simplify the edge cases and test more values. The tensors can be of size 4096, for example.

Reply:
>For the NumPy comparison start a new test and set the default scalar type to double in a try/finally block and test uint8, int8, long, float32, and complex64 dtype division.

Split this into 2 tests. `div_rounding_nonfinite` to test fp special values and `div_rounding_numpy` to test division of random tensors. Note that the rounding modes don't support complex. This is consistent with Python, e.g.:
```python
>>> 1j // 2
TypeError: can't take floor of complex number.
```

Although, I'm not sure what you meant by 

> Enumerate the modes on large random tensors [...]  **and then every other element of those tensors**

Not sure what you mean in the second part. Do you mean `divide(a[None, :], b[:, None], ...)` so I'm testing 4096 * 4096 combinations?

Reply:
> @mruberry I'm not sure what you meant by the second part. Do you mean `divide(a[None, :], b[:, None], ...)` so I'm testing 4096 * 4096 combinations?

Sorry, that was unclear. I was thinking:

```
a = ...
a_discontiguous = a[::2]
```

Then a_discontiguous will have every other element of a. Does that make sense?

Reply:
Oh I see, "every second element" not "all other elements".